object_store/src/lib.rs

```diff
@@ -718,25 +718,41 @@ pub struct GetOptions {
 
 impl GetOptions {
     /// Returns an error if the modification conditions on this request are not satisfied
-    fn check_modified(
-        &self,
-        location: &Path,
-        last_modified: DateTime<Utc>,
-    ) -> Result<()> {
-        if let Some(date) = self.if_modified_since {
-            if last_modified <= date {
-                return Err(Error::NotModified {
-                    path: location.to_string(),
-                    source: format!("{} >= {}", date, last_modified).into(),
+    ///
+    /// <https://datatracker.ietf.org/doc/html/rfc7232#section-6>
+    fn test(&self, meta: &ObjectMeta) -> Result<()> {
```

Comment:
```suggestion
    fn check_modification_conditions(&self, meta: &ObjectMeta) -> Result<()> {
```

"test" is rather non-descriptive and is easily confused w/ unit tests.